hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientClusterProxy.java

```diff
@@ -127,6 +131,11 @@ public void changeClusterVersion(@Nonnull Version version, @Nonnull TransactionO
         throw new UnsupportedOperationException();
     }
 
+    @Override
+    public boolean isEnterprise() {
+        return this.clusterService.getMemberList().size() > 0 && this.connectionManager.isConnectedToEnterpriseCluster();
```

Comment:
why do you check member list size? I think that this should just return 
```
    return clusterService.isEnterprise();
```
And the cluster service should handle the rest. This also eliminates the new extra dependency you introduced the connection manager on this class.

Reply:
The isFailoverSupported information is in the connection manager. I think we can add a setter "setIsEnterprise()" to the cluster proxy and call it in the connection manager instead of keeping a local variable.

Then this cluster proxy would just return it here. 

However, I think we need to know whether we are connected or not because my method returns whether we are connected to a EE cluster. 

Maybe we can use the setter I mentioned to set "isEnterprise" to false if we disconnect.